### PR TITLE
Readme update

### PR DESCRIPTION
**Pull request by Domejko**
Added **Contributing** section to `README.md`

### CODE REVIEW

**Comment by alat-rights** [comment]
Thanks for making the PR.

Nits:
1. We should mention that we use the [Standard](https://github.com/microsoft/pyright/issues/6607) mode on PyRight.
2. I would probably do without the preamble. It's not a great fit for a project of this (small) size.

BTW: I think this project will have more detailed workflow soon. E.g. I think I'm going to have separate `dev` and `release` branches. There will be some focus on making the project stable.

For future reference: the more minimal a PR is, the easier it is for us to get it merged.

**Comment by alat-rights** [comment]
#21

**Comment by Domejko** [comment]
> 1. We should mention that we use the [Standard](https://github.com/microsoft/pyright/issues/6607) mode on PyRight.

But there is no flag to differentiate `basic/standard/strict` and we are not using neither `pyproject.toml` or `pyrightconfig.json` for configuration, so it's used with default settings. Then I think it's not necessary to mention it.



> 2\. I would probably do without the preamble. It's not a great fit for a project of this (small) size.

I will reduce it to simple greeting then. To be nice and welcoming is always good.

File: docs/README.md [hunk 1]
```diff
@@ -106,7 +106,29 @@ The documentation for this project is hosted at [utils.alana.computer](https://u
   - `alana.remove_xml` to strip certain XML tag-enclosed content from a string (along with the tags). This is primarily intended to get rid of "<reasoning>...</reasoning>" strings. ⚠️ Regex parsing of XML may be unreliable!
 - A bunch of aliases (Try: `alana.few_shot`, `alana.n_shot`, or `alana.xml`)
 
-## Testing
+## Contributing
+A big welcome and thank you for considering contributing to alana-utilities open source project!
```
Comment (alat-rights) [inline docs/README.md:110]:
```suggestion
Welcome! Thank you for considering making a contribution to `alana-utilities`.
```

File: docs/README.md [hunk 2]
```diff
@@ -106,7 +106,29 @@ The documentation for this project is hosted at [utils.alana.computer](https://u
...
+
+### Getting started
+For changes that address core functionality or would require breaking changes (e.g. a major release), it's best to open an Issue to discuss your proposal first. This is not required but can save time creating and reviewing changes.
```
Comment (alat-rights) [inline docs/README.md:113]:
```suggestion
If you'd like to make a major refactor or breaking change, please open an [Issue](https://github.com/alat-rights/alana-utilities/issues) first. This way, I can review it to determine if it is within scope for this project.

In general, there are no guarantees that your pull requests will be merged into the library. `alana-utilities` is essentially a hobby project, and there is currently only one maintainer.
```

File: docs/README.md [hunk 3]
```diff
@@ -106,7 +106,29 @@ The documentation for this project is hosted at [utils.alana.computer](https://u
...
+In general, we follow the ["fork-and-pull" Git workflow](https://github.com/susam/gitpr).
+
+### Style and formating
+We are using [Black](https://github.com/psf/black) for formating and [PyRight](https://github.com/microsoft/pyright) for type checking.
+
+Before pushing your code and making PR make sure that you have run `black .` on your code, otherwise it will fail.
```
Comment (alat-rights) [inline docs/README.md:119]:
```suggestion
Before submitting a pull request, please run `black .` on your code. This saves us from failed CI runs.
```

File: docs/README.md [hunk 4]
```diff
@@ -106,7 +106,29 @@ The documentation for this project is hosted at [utils.alana.computer](https://u
...
+To install and run black use this commands:
```
Comment (alat-rights) [inline docs/README.md:120]:
```suggestion
To install and run `Black`:
```

File: docs/README.md [hunk 5]
```diff
@@ -106,7 +106,29 @@ The documentation for this project is hosted at [utils.alana.computer](https://u
...
+```commandline
+  $ pip install black
+  $ black .
+```
+To keep types in code consistent:
```
Comment (alat-rights) [inline docs/README.md:125]:
```suggestion
To install and run `PyRight`:
```

**Review by alat-rights — CHANGES_REQUESTED** [review]
Should be good with these changes.

File: docs/README.md [hunk 6]
```diff
@@ -106,7 +106,31 @@ The documentation for this project is hosted at [utils.alana.computer](https://u
...
 - A bunch of aliases (Try: `alana.few_shot`, `alana.n_shot`, or `alana.xml`)
 
-## Testing
+## Contributing
+Welcome! Thank you for considering making a contribution to `alana-utilities`.
+
+### Getting started
+If you'd like to make a major refactor or breaking change, please open an [Issue](https://github.com/alat-rights/alana-utilities/issues) first. This way, I can review it to determine if it is within scope for this project.
+
+In general, there are no guarantees that your pull requests will be merged into the library. `alana-utilities` is essentially a hobby project, and there is currently only one maintainer.
```
Comment (alat-rights) [inline docs/README.md:115]:
```suggestion
There are no guarantees that your pull requests will be merged into the library. `alana-utilities` is essentially a hobby project, and there is currently only one maintainer.

```

File: docs/README.md [hunk 7]
```diff
@@ -106,7 +106,31 @@ The documentation for this project is hosted at [utils.alana.computer](https://u
...
+### Style and formating
+We are using [Black](https://github.com/psf/black) for formating and [PyRight](https://github.com/microsoft/pyright) for type checking.
+
+Before submitting a pull request, please run `black .` on your code. This saves us from failed CI runs.
+To install and run `Black`:
+```commandline
+  $ pip install black
+  $ black .
+```
+To install and run `PyRight`:
```
Comment (alat-rights) [inline docs/README.md:127]:
```suggestion

To install and run `PyRight`:
```

File: docs/README.md [hunk 8]
```diff
@@ -106,7 +106,31 @@ The documentation for this project is hosted at [utils.alana.computer](https://u
...
+Welcome! Thank you for considering making a contribution to `alana-utilities`.
+
+### Getting started
+If you'd like to make a major refactor or breaking change, please open an [Issue](https://github.com/alat-rights/alana-utilities/issues) first. This way, I can review it to determine if it is within scope for this project.
+
+In general, there are no guarantees that your pull requests will be merged into the library. `alana-utilities` is essentially a hobby project, and there is currently only one maintainer.
+In general, we follow the ["fork-and-pull" Git workflow](https://github.com/susam/gitpr).
+
+### Style and formating
+We are using [Black](https://github.com/psf/black) for formating and [PyRight](https://github.com/microsoft/pyright) for type checking.
```
Comment (alat-rights) [inline docs/README.md:119]:
```suggestion
We are using [Black](https://github.com/psf/black) for formatting and [PyRight](https://github.com/microsoft/pyright) for type checking.
```

File: docs/README.md [hunk 9]
```diff
@@ -106,7 +106,31 @@ The documentation for this project is hosted at [utils.alana.computer](https://u
...
+
+Before submitting a pull request, please run `black .` on your code. This saves us from failed CI runs.
+To install and run `Black`:
```
Comment (alat-rights) [inline docs/README.md:122]:
```suggestion

To install and run `Black`:
```

**Review by alat-rights — DISMISSED** [review]
Just caught a few more minor things :)

**Pull request merged**